app/controllers/api/v1/subnets_controller.rb

```diff
@@ -25,6 +25,7 @@ def show
       api :POST, '/subnets', 'Create a subnet'
       param :subnet, Hash, :required => true do
         param :name, String, :desc => 'Subnet name', :required => true
+        param :type, String, :desc => 'Type or protocol, Subnet::Ipv4 or Subnet::Ipv6', :required => true
```

Comment:
I agree, it should be optional. Good catch.
The default value should be set via STI-Magic. The controller actually creates a new instance of "Subnet", which is then casted to either "Subnet::Ipv4" if type is not set. As I wrote in the initial PR comment, i don't know if this is a good idea, though.